plugin/session_buffer.py

```diff
@@ -322,10 +322,12 @@ def on_post_save_async(self, view: sublime.View, new_uri: DocumentUri) -> None:
             if send_did_save:
                 self.purge_changes_async(view)
                 self.session.send_notification(did_save(view, include_text, self.last_known_uri))
-        if self.should_show_diagnostics_panel:
-            mgr = self.session.manager()
-            if mgr:
+        mgr = self.session.manager()
+        if mgr:
+            if self.should_show_diagnostics_panel:
                 mgr.show_diagnostics_panel_async()
+            elif userprefs().show_diagnostics_panel_on_save > 0:
+                mgr.hide_diagnostics_panel_async()
```

Comment:
Actually I've realized now that this logic is not really correct when there is more than one server running on a file. In that case one server can hide the panel while another server might still have relevant diagnostics.

This logic should rather be around here

https://github.com/sublimelsp/LSP/blob/30a8aac1816c02521d96fc5aa2ffdc6a8e0c66b7/plugin/documents.py#L280-L284

and/or

https://github.com/sublimelsp/LSP/blob/30a8aac1816c02521d96fc5aa2ffdc6a8e0c66b7/plugin/documents.py#L369-L385